### PR TITLE
Adding time supplier function

### PR DESCRIPTION
**Pull request by sobychacko**

### CODE REVIEW

File: supplier/time-supplier/src/test/java/io/pivotal/java/function/time/supplier/TimeSupplierApplicationTests.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Soby Chacko
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class TimeSupplierApplicationTests {
```
Comment (dturanski) [inline supplier/time-supplier/src/test/java/io/pivotal/java/function/time/supplier/TimeSupplierApplicationTests.java]: Should have some additional tests with valid/invalid DateFormat

File: supplier/time-supplier/src/main/java/io/pivotal/java/function/time/supplier/DateFormat.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import javax.validation.Constraint;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import javax.validation.Payload;
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.text.SimpleDateFormat;
+
+import static java.lang.annotation.ElementType.*;
```
Comment (artembilan) [inline supplier/time-supplier/src/main/java/io/pivotal/java/function/time/supplier/DateFormat.java]: No asterisk imports, please, at all.

File: supplier/time-supplier/src/main/java/io/pivotal/java/function/time/supplier/DateFormat.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
```
Comment (artembilan) [inline supplier/time-supplier/src/main/java/io/pivotal/java/function/time/supplier/DateFormat.java]: No `static` imports in production classes, please, at all

File: supplier/time-supplier/src/main/java/io/pivotal/java/function/time/supplier/DateFormat.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+/**
+ * The annotated String must be a valid {@link java.text.SimpleDateFormat} pattern.
+ *
+ * @author Eric Bottard
```
Comment (artembilan) [inline supplier/time-supplier/src/main/java/io/pivotal/java/function/time/supplier/DateFormat.java]: Oh! I see this is copy/paste from some other place.

@dturanski , you see this is what I have talked about new projects and their relaxed attitude for code style.

File: supplier/time-supplier/src/main/java/io/pivotal/java/function/time/supplier/DateFormat.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+	/**
+	 * Defines several {@link DateFormat} annotations on the same element.
+	 *
+	 * @see DateFormat
+	 */
+	@Target({METHOD, FIELD, ANNOTATION_TYPE, CONSTRUCTOR, PARAMETER})
+	@Retention(RUNTIME)
+	@Documented
+	@interface List {
```
Comment (artembilan) [inline supplier/time-supplier/src/main/java/io/pivotal/java/function/time/supplier/DateFormat.java]: I think we can live without this one.
I'm surprised though that Hibernate Validator doesn't have a date format validator out-of-the-box 😄

File: supplier/time-supplier/src/main/java/io/pivotal/java/function/time/supplier/TimeSupplierConfiguration.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+/**
+ * @author Soby Chacko
+ */
+@Configuration
+@EnableConfigurationProperties(TimeProperties.class)
+public class TimeSupplierConfiguration {
+
+	@Bean
+	public Supplier<String> timeSupplier(TimeProperties timeProperties) {
+		return () -> new SimpleDateFormat(timeProperties.getDateFormat()).format(new Date());
```
Comment (artembilan) [inline supplier/time-supplier/src/main/java/io/pivotal/java/function/time/supplier/TimeSupplierConfiguration.java]: Let's consider to use a `FastDateFormat` instead and just have its single instance for the whole function, but not `new SimpleDateFormat` for every supplier call!

**Comment by artembilan** [comment]
Merged as https://github.com/pivotal/java-functions/commit/621e5074ae71708b8afac83735650983cae68d8b

**Pull request closed without merging**